lib/parliament/request/base_request.rb

```diff
@@ -1,78 +1,83 @@
       #
-      #   Parliament::Request.new(base_url: 'http://example.com').base_url #=> 'http://example.com'
+      #   Parliament::Request::BaseRequest.new(base_url: 'http://example.com').base_url #=> 'http://example.com'
       #
       # @example Overriding the base_url via a parameter
       #   ENV['PARLIAMENT_BASE_URL'] #=> 'http://test.com'
       #
-      #   Parliament::Request.new(base_url: 'http://example.com').base_url #=> 'http://example.com'
+      #   Parliament::Request::BaseRequest.new(base_url: 'http://example.com').base_url #=> 'http://example.com'
       #
       # @example Setting the headers on the class
-      #   Parliament::Request.headers = { 'Accept' => 'Test' }
+      #   Parliament::Request::BaseRequest.headers = { 'Accept' => 'Test' }
       #
-      #   Parliament::Request.new.headers #=> '{ 'Accept' => 'Test' }
+      #   Parliament::Request::BaseRequest.new.headers #=> '{ 'Accept' => 'Test' }
       #
       # @example Setting the headers via a parameter
-      #   Parliament::Request.headers #=> nil
+      #   Parliament::Request::BaseRequest.headers #=> nil
       #
-      #   Parliament::Request.new(headers: '{ 'Accept' => 'Test' }).headers #=> { 'Accept' => 'Test' }
+      #   Parliament::Request::BaseRequest.new(headers: '{ 'Accept' => 'Test' }).headers #=> { 'Accept' => 'Test' }
       #
       # @example Overriding the headers via a parameter
-      #   Parliament::Request.headers = { 'Accept' => 'Test' }
+      #   Parliament::Request::BaseRequest.headers = { 'Accept' => 'Test' }
       #
-      #   Parliament::Request.new(headers: '{ 'Accept' => 'Test2' }).headers #=> { 'Accept' => 'Test2' }
+      #   Parliament::Request::BaseRequest.new(headers: '{ 'Accept' => 'Test2' }).headers #=> { 'Accept' => 'Test2' }
       #
       # @param [String] base_url the base url of our api. (expected: http://example.com - without the trailing slash).
       # @param [Hash] headers the headers being sent in the request.
-      def initialize(base_url: nil, headers: nil, builder: nil)
+      # @param [Parliament::Builder] builder the builder to use in order to build a response.
+      # @params [Module] decorators the decorator module to use in order to provide possible alias methods for any objects created by the builder.
+      def initialize(base_url: nil, headers: nil, builder: nil, decorators: nil)
         @base_url = base_url || self.class.base_url
         @headers = headers || self.class.headers || {}
         @builder = builder || Parliament::Builder::BaseResponseBuilder
+        @decorators = decorators
       end
 
-      # Using our url built via #method_missing, make a HTTP GET request and process results into a response.
+      # Makes an HTTP GET request and process results into a response.
       #
       # @example HTTP GET request
-      #   request = Parliament::Request.new(base_url: 'http://example.com')
-      #
-      #   # url: http://example.com/people/123456
```

Comment:
Should the URL lines be kept in for completeness? Especially in the blow examples where parameters are used and it's less clear where we are getting?